go-controller/pkg/util/libovsdb.go

```diff
@@ -70,21 +70,46 @@ func newClient(cfg config.OvnAuthConfig, dbModel *model.DBModel, onReconnectFn R
 }
 
 // NewSBClient creates a new OVN Southbound Database client
-func NewSBClient(onReconnectFn ReconnectFunc, stopCh <-chan struct{}) (client.Client, error) {
+func NewSBClient(stopCh <-chan struct{}) (client.Client, error) {
+	return NewSBClientWithConfig(config.OvnSouth, stopCh)
+}
+
+// NewSBClient creates a new OVN Southbound Database client with the provided configuration
+func NewSBClientWithConfig(cfg config.OvnAuthConfig, stopCh <-chan struct{}) (client.Client, error) {
 	dbModel, err := sbdb.FullDatabaseModel()
 	if err != nil {
 		return nil, err
 	}
-	return newClient(config.OvnSouth, dbModel, onReconnectFn, stopCh)
+	onConnectFn := func(client client.Client) error {
+		// add event handlers as needed
+		// client.Cache().AddEventHandler(...)
```

Comment:
There is a problem with this :cry: 
In rebasing #2262 I can't add an EventHandler here, because the `util` function would need access to the unidling controllers event queue, which isn't easy. I'm not sure how we solve this outside of not requiring a reconnetFn :thinking: